### ISSUE TITLE
Codecov integration reporting inaccurate results

### ISSUE THREAD

**Issue by KPrasch** [nucypher/nucypher#1524]
Each circleCI container is pushing it's own `reports` directory directly to codecov without any strategy for combining the reports from other jobs.  This results in the dreaded red "X", and inaccurate patch results on PRs. 

It *might* be possible to mitigate by using circleCI workspaces to persist reports until the end of a workflow and *then* combine them before uploading... but I'm uncertain what a workable approach is.

**Comment by cygnusv** [comment]
Yes, I absolutely agree we need to improve the codecov step. Also, sometimes it takes a long time (or even fails) to upload the results. I think doing it once per workflow would be probably for the best.

**Comment by cygnusv** [comment]
#1748 modifies codecov execution so it's only performed in the nightly build. This is not solving the problem in the original issue, it's just hiding it under the rug (although we've effectively halved the overall CI build time)
